src/gui/dialogs/preferences_dialog.cpp

```diff
@@ -638,6 +655,12 @@ void tpreferences::initialize_members(twindow& window)
 			boost::ref(textbox),
 			boost::ref(window)));
 
+	friend_list.set_callback_value_change(boost::bind(
```

Comment:
IMO that should be handled in the post_show function. The display function isn't meant to be implemented outside the header.